components/prism-cooklang.js

```diff
@@ -0,0 +1,152 @@
+(function (Prism) {
+
+	var single_token_suffix = /(?:(?!\s)[\d$+<=a-zA-Z\x80-\uFFFF])+/.source;
+	var multi_token_infix = /[^{}@#]+/.source;
+	var multi_token_suffix = /\{[^}#@]*\}/.source;
+
+	var multi_token = multi_token_infix + multi_token_suffix;
+
+	var timer_units = /(?:h|hours|hrs|m|min|minutes)/.source;
+
+	var amount_group_impl = {
+		pattern: new RegExp(/\{/.source
+			+ '(?:(?:'
+			+ /(?:[^{}|*%]+\*?)/.source // optional serving scale
+			+ '|'
+			+ /(?:(?:[^{}|*%]+\|)+[^{}|*%]+)/.source // serving alternatives
+			+ ')' + /(?:%[^{}|*%]+)?/.source //optional unit
+			+ ')?'
+			+ /\}/.source
+		),
```

Comment:
This pattern should not be so strict. The `multi_token_suffix` pattern allows a lot, so this pattern being so strict will cause what you can see in `timer_feature.test`: 

```json
	["ingredient-group", [
		["ingredient-keyword", "@"],
		["ingredient", "milk"],
		"{3%*%L}"
	]],
```

In general, it's better to make the parent pattern (in this case `multi_token_suffix`) stricter than the inside pattern (in this case `amount_group_impl`). 

However, I think that it might better to just allow pretty much anything inside braces. Prism isn't a parser. We can highlight syntactically incorrect code however we want. We typically even want to highlight some invalid code to give a better experience while typing.

```suggestion
		pattern: /\{[^{}]*\}/,
```